controllers/hasapplication_controller_conditions.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+)
+
+func (r *HASApplicationReconciler) SetCreateConditionAndUpdateCR(ctx context.Context, hasApplication *appstudiov1alpha1.HASApplication, createError error) {
```

Comment:
would you mind updating the param to runtime.Object from appstudiov1alpha1.HASApplication? 

I can change the logic inside to accomodate the other CRs with my branch